pandas/_typing.py

```diff
@@ -256,7 +256,7 @@ def closed(self) -> bool:
 # compression keywords and compression
 CompressionDict = Dict[str, Any]
 CompressionOptions = Optional[
-    Union[Literal["infer", "gzip", "bz2", "zip", "xz", "zstd"], CompressionDict]
+    Union[Literal["infer", "gzip", "bz2", "zip", "xz", "zstd", "tar"], CompressionDict]
```

Comment:
Does tar.gz, tar.z2, ... work?

Reply:
yes! the `tarfile` standard library we're using has support for either `.gz`, `.z2` etc: https://docs.python.org/3/library/tarfile.html
since this `CompressionOptions` dict only specifies the pandas-internal decompression mechanism to use, we don't need to list any other extensions here.